push/sender/src/test/java/org/jboss/aerogear/unifiedpush/message/TestMetricsCollector.java

```diff
@@ -130,13 +129,12 @@ public void test(PushMessageInformationDao pushMessageInformationDao) throws Int
         assertNull(receive().withTimeout(100).withSelector("variantID = '%s'", variantID2+":"+pushMetric.getId()).from(allBatchesLoaded));
     }
 
-    private VariantMetricInformation updateVarianMetricById(PushMessageInformation pmi, VariantMetricInformation vmi) {
-        for (VariantMetricInformation v : pmi.getVariantInformations()) {
-            if (vmi.getVariantID().equals(v.getVariantID())) {
-                return v;
-            }
-        }
-        throw new IllegalStateException("No variant metric found for given id");
+    private VariantMetricInformation updateVarianMetricById(PushMessageInformation pmi, final VariantMetricInformation vmi) {
+
+        return pmi.getVariantInformations().stream()
+                .filter(v -> vmi.getVariantID().equals(v.getVariantID()))
+                .findFirst()
+                .orElse(null);
```

Comment:
If we aren't throwing the illegal state exception any more we should consider making the return type Optional<VariantMetricInformation>